### DISCUSSION TITLE
Final cumulative incidence in risk table

### DISCUSSION THREAD

**Discussion by juliatubert** [pharmaverse/ggsurvfit#232] [Q&A]
Hello,

Thanks for this wonderful package! There is one thing I'm stuck on, however.

Sometimes in my work, the maximum follow-up time is not a nice/round number or it may be different between strata, but I'd still like to be able to show the final cumulative incidence reflecting the total cumulative events. For example, in the figure below, max follow-up time is 4.67 months, but the final cumulative incidence is displayed at 5 months.
![image](https://github.com/user-attachments/assets/791e5704-0664-4bb6-8ace-d77532f7797c)

I can't figure out how to do this with ggsurvfit. Is there a way to "force in" the final cumulative incidence at 25 months in the example below?
```{r}
library(ggsurvfit)
survfit2(Surv(ttdeath, death_cr) ~ trt, tidycmprsk::trial) |> 
  ggcuminc(outcome = "death from cancer") +
  add_risktable(
  risktable_stats =
    c("{n.risk}",
    "{cum.event}",
    "{round((estimate)*100)}%"),
  stats_label = c("At Risk", 
                  "Cumulative Events",
                  "Cumulative Incidence"),
  size = 3) +
  scale_ggsurvfit(x_scales = list(breaks = seq(0, 25, by = 5),
                                  limits = c(-1, 26))) #set limits to avoid text cutoff
```
![image](https://github.com/user-attachments/assets/0396a9e2-dfcc-4009-8975-83c098270782)

Thank you!

**Comment by ddsjoberg** [comment]
Hmmmm, the estimate is not defined at time = 25, which is why it's printing as NA. I don't think there is a way around that.

You could use `tidycmprsk::tbl_cuminc()` to have a separate table with the results?

**Reply by juliatubert** [reply]
Thanks so much for your quick response!

I played around with tidycmprsk::tbl_cuminc() and time= in add_risktable(), but wasn't able to get the result I was going for.

I was able to come up with a workaround by using conditional logic to replace the missing estimates with max cumulative incidence. It's not the most elegant solution, so other suggestions or improvements to my code are still welcome. Thanks!

```{r}
# Generate survival/cumulative incidence data
library(ggsurvfit)
cuminc <- survfit2(Surv(ttdeath, death_cr) ~ trt, tidycmprsk::trial) |> 
  ggcuminc(outcome = "death from cancer")

# Pull maximum cumulative incidence
max_cuminc <- cuminc$data |> group_by(strata) |> summarize(max_cuminc=round((max(estimate))*100))
max_cumincA <- max_cuminc |> filter(strata=='Drug A') |> pull(max_cuminc)
max_cumincB <- max_cuminc |> filter(strata=='Drug B') |> pull(max_cuminc)

# Plot
cuminc +
 add_risktable(
  risktable_stats =
    c("{n.risk}",
    "{cum.event}",
    #"{round((estimate)*100)}%"), #NA for last estimate
    "{case_when(!is.na(estimate) ~ round(estimate*100),
                is.na(estimate) & strata=='Drug A' ~ max_cumincA,
                is.na(estimate) & strata=='Drug B' ~ max_cumincB)}%"), #workaround
  stats_label = c("At Risk", 
                  "Cumulative Events",
                  "Cumulative Incidence"),
  size = 3
) +
  scale_ggsurvfit(x_scales = list(breaks = seq(0, 25, by = 5),
                                  limits = c(-1, 26))) #set limits to avoid text cutoff
```
![image](https://github.com/user-attachments/assets/f7743ced-239d-4822-beb8-dd8315907ac0)
